src/models/pretraining.py

```diff
@@ -0,0 +1,99 @@
+import torch.nn as nn
+import torch
+from transformers import BertModel, BertConfig
+import numpy as np
+from torch.nn import CTCLoss
+import lpips
+
+from src.utils.common import PositionalEncoding, compute_ctc_loss
+from src.utils.masking import SpanMaskingGenerator
+from src.models.vtr.ocr import OCRHead
+
+
+class Pretrain(nn.Module):
+    def __init__(
+        self,
+        emb_size: int = 512,
+        n_head: int = 8,
+        n_layers: int = 4,
+        device: str = "cpu",
+        ocr: OCRHead = None,
+        char2int_dict: dict = None,
+        alpha: float = 1,
+    ):
+        super().__init__()
+        config = BertConfig(hidden_size=emb_size, num_attention_heads=n_head)
+        self.encoder = BertModel(config)
```

Comment:
I expected that we would use [TransformerEnocder](https://pytorch.org/docs/stable/generated/torch.nn.TransformerEncoder.html) instead of adding a dependency from HF